docs/concepts/openedx-events.rst

```diff
@@ -0,0 +1,93 @@
+
+Workflow
+~~~~~~~~
+
+The workflow of emitting and processing an Open edX Event with N receivers is as follows:
+
+The event execution process follows these steps:
+
+#. An application (caller) emits an event by calling the `send_event`_ method implemented by `OpenEdxPublicSignal`_ which the event inherits from.
+
+#. The caller passes the :term:`event payload<Event Payload>` to the `send_event`_ method. The event payload is the data associated with the event that is passed to the receivers when it's triggered, and uses data attribute classes (e.g. ``CourseEnrollmentData``, ``UserData``, etc.) to carry data about the event.
+
+#. The `send_event`_ method generates Open edX-specific metadata for the event on the fly, like the event version or the timestamp when the event was sent. The event receivers can access this metadata during their processing.
+
+#. After, the `send_event`_ method calls the `send or send_robust`_ method from Django Signals under the hood. The ``send`` method is used for development and testing, while the ``send_robust`` method is used in production to ensure receivers don't raise exceptions halting the application process.
+
+#. Building on Django Signals allows us to use the same `Django signals registry mechanism`_ for receiver management. This means that developers can register `signal receivers in their plugins`_ for Open edX Events in the same way they would for Django signals.
+
+#. All registered receivers are executed in the order they were registered. Each receiver processes the event data and performs the necessary actions.
+
+#. After all receivers for the event have been executed, the process continues with the application logic.
+
+Here is an example of an event that saves a user's notification preferences when they enroll in a course:
+
+#. A user enrolls in a course, `triggering the COURSE_ENROLLMENT_CREATED event`_. This event includes information about the user, course, and enrollment details.
+
+#. A signal receiver listening for the ``COURSE_ENROLLMENT_CREATED`` event is called and processes the event data.  In this case, it would be the `course_enrollment_post_save receiver`_.
+
+#. The signal receiver creates a notification preference for the user, enabling them to receive notifications about the course.
+
+#. The application continues with the course enrollment process.
+
+The `Django Signals Documentation`_ provides a more detailed explanation of how Django signals work.
+
+How are Open edX Events used?
+-----------------------------
+
+Developers can listen to Open edX Events by registering signal receivers from their Open edX Django plugins that respond to the emitted events. This is done using Django's signal mechanism, which allows developers to listen for events and execute custom logic in response.
+
+For more information on using Open edX Events, refer to the `Using Open edX Events`_ how-to guide.
```

Comment:
Do you want to reference the doc here using a `:ref:` directive?

Reply:
I used `:doc:` instead, does that work?

Reply:
John Swope told me that `:ref:` is better, but I think `:doc:` is reasonable (`:ref:` works if the doc filepath changes, but `:doc:` works if the title of the doc or the reference string changes, so, they both have drawbacks)